docs/source/dataset.rst

```diff
@@ -0,0 +1,234 @@
+is young and quickly evolving. There exists multiple formats for stac item. Only a very specific format 
+of stac item is supported by geo-deep-learning. If using stac items with geo-deep-learning, make sure 
+they follow the structure of our *Worldview-2* example above. 
+
+Bands must be selected by `common name <https://github.com/stac-extensions/eo/#common-band-names>`_ 
+in dataset config:
+
+.. code-block:: yaml
+
+    dataset:
+        [...]
+        bands: ["red", "green", "blue"]
+
+.. note::
+    Order matters, ``["red", "green", "blue"]`` is not equal to ``["blue", "green", "red"]`` !
+
+
+Under the hood
+--------------
+
+For use cases *Multi-band raster files with more bands or different order than needed* to 
+*Single-band raster files, identified as assets in a stac item*, geo-deep-learning creates 
+a `virtual raster <https://gdal.org/drivers/raster/vrt.html>`_ to bring the variable source 
+raster data to a common destination format without rewriting existing data to disk.
+
+.. _yamlparameters:
+
+Dataset configuration yaml file
+-------------------------------
+
+.. literalinclude:: ../../config/dataset/test_ci_segmentation_multiclass.yaml
+   :language: yaml
+
+- ``name`` (str)
+    Name of the dataset.
+- ``raw_data_csv`` (str)
+    Path to csv referencing source data for tiling or training.
+- ``raw_data_dir`` (str)
+    The directory where online data may be downloaded.
+- ``download_data`` (bool)
+    If True, all source data as url will be downloaded to raw data directory. 
+    The local copy will be used afterwards.
+- ``bands`` (list)
+    Bands to be selected during the tiling process. Order matters 
+    (ie ``["red", "green", "blue"]`` is not equal to ``["blue", "green", "red"]``). See
+    :ref:`single or multi bands imagery <datasetsinglemultiband>` section for more 
+    information on input data and band selection and ordering.
+- ``attribute_field`` (str)
+    Name of the attribute from the ground truth data.
+- ``attribute_values`` (list)
+    Filter only relevant classes from the ground truth data by listing the value associated to 
+    the class you desire.
+    For example, if a ground truth GeoPackage contains polygons belonging to 4 classes of 
+    interests (forests, water bodies, roads, buildings), a user can filter out all non-building 
+    polygon by choosing an attribute field and value that separate building polygons from 
+    others. Though attribute values may not be continuous numbers starting at 1 
+    (0 being background), Geo-deep-learning ensures all values during training are continuous
+    and, therefore, match values from predictions.
+- ``class_name`` (str)
+    Will follow soon.
+- ``classes_dict`` (dict)
+    Dictionary containing the name of the class and the value associated to them.
+- ``class_weights`` (dict)
+    Dictionary containing the class value and the minimum percentage that must be include 
+    in the tiling dataset.

```

Comment:
The class_weights is used as a balancing strategy during training and is implemented at the loss level. I don't think "minimum percentage [...] in the tiling dataset" describes accuratly what this parameter is used for. 
See:
https://github.com/NRCan/geo-deep-learning/blob/ef96eb6c77dd031c01f981790970bcd93456286b/utils/loss.py#L16